plugins/task-plugin/src/che-workspace-client.ts

```diff
@@ -109,7 +109,10 @@ export class CheWorkspaceClient {
     if (!machineExecServer) {
       throw new Error(`No server with type ${TERMINAL_SERVER_TYPE} found.`);
     }
-    return machineExecServer.url!;
+    if (!machineExecServer.attributes!.port) {
+      throw new Error('No machine-exec-server attributes.port found');
+    }
+    return `ws://127.0.0.1:${machineExecServer.attributes!.port}`;
```

Comment:
I wonder if terminal does not have the same issues you faced with tasks.

Reply:
Ah, after taking a deeper look I think terminal does not have such issue because it's a browser (not theia backend) who does certificates validation.

And I see some benefit in using localhost for tasks, but I don't see it as a solution for https://che-incubator.github.io/2021/02/01/@mario.loriedo-using-mkcert-to-locally-trust-eclipse-che-tls-certificates-ffaafe76e5d0.html
I assume that the issues caused by fact that Theia Backend needs CA certificate. See https://github.com/eclipse/che-theia/pull/1019/files#diff-fdc678251442b2044dd33d56c0a5ff4498f17c42b005cd63a696eb46f5703c81L129
But there may be some bug and end-certificate is propagated to Theia backend as well.
https://github.com/eclipse/che-operator/blob/master/pkg/deploy/server/deployment_che.go#L63

If you're able to share what's is propagate inside Che Theia - we would be able to check the theory above.